calculator1.rb

```diff
@@ -0,0 +1,110 @@
+def prompt_until_numeric(number)
+  until  number == number.to_i.to_s || number == number.to_f.to_s
+    puts "Whoops that's not a number, try again!"
+    number = gets.chomp
+  end
+  return number
+end
+#can change to deal with integer or float, make if/else and return number.to_i
+#return number.to_f
+
+def prompt_until_numeric_and_nonzero(number)
+until  (number == number.to_i.to_s || number == number.to_f.to_s) && number.to_i != 0
+    puts "Whoops that's not a number we can divide by, try again!"
+    number = gets.chomp
+  end
+  return number
+end
+
+def addition(first_number, second_number)
+  return first_number + second_number
+end
+
+def subtraction(first_number, second_number)
+  return first_number - second_number
+end
+
+def multiplication(first_number, second_number)
+  return first_number * second_number
+end
+
+ def division(first_number, second_number)
+     return first_number/second_number
+ end
+
+ def exponential(first_number, second_number)
+   return first_number ** second_number
+ end
+
+ def modulo_calculation(first_number, second_number)
+   return first_number % second_number
+ end
+
+puts "Welcome to my calculator!"
+puts "What would you like to do?"
+operators = ["add", "+", "substract","-", "multiply", "*", "divide", "/", "exponent", "**", "^", "modulo", "%"]
+operators_prompt =["add(+)", "substract(-)", "multiply(*)", "divide(/)", "exponent(** or ^)", "modulo(%)"]
+i = 1
+operators_prompt.each do |operator|
+  puts "#{i}: #{operator}"
+  i += 1
+end
+puts "Please choose one operator (name or symbol):"
+operator_selected = gets.chomp.downcase.to_s
+until operators.include?(operator_selected)
+  puts "Please enter 'add', 'substract', 'multiply', or 'divide'."
+  operator_selected = gets.chomp.downcase.to_s
+end
+
+case operator_selected
+when "add", "+"
+    puts "We're doing addition!"
+    return operator_selected = "+"
```

Comment:
I'm looking at addition, subtraction and multiplication, and I don't think that they will finish the program the way you intended. It would be worth your time to review them and see if you can figure out how to fix them. If you have trouble, feel free to ask me for guidance.